tests/test_operation.py

```diff
@@ -451,6 +451,35 @@ class MyOp(qml.operation.Operator):
         assert MyOp.has_diagonalizing_gates is False
         assert MyOp(wires=0).has_diagonalizing_gates is False
 
+    def test_has_generator_true(self):
+        """Test `has_generator` property detects overriding of `generator` method."""
+
+        class MyOp(qml.operation.Operator):
+            num_wires = 1
+
+            @staticmethod
+            def generator():
+                return np.eye(2)
+
+        assert MyOp.has_generator is True
+        assert MyOp(wires=0).has_generator is True
+
+    def test_has_generator_false(self):
+        """Test `has_generator` property defaults to false if `generator` not overwritten."""
+
+        class MyOp(qml.operation.Operator):
+            num_wires = 1
+
+        assert MyOp.has_generator is False
+        assert MyOp(wires=0).has_generator is False
+
+    def test_has_generator_false_concrete_template(self):
```

Comment:
Perhaps a bit redundant but maybe an analogous test for a concrete operation that _does_ have a generator defined could be added? E.g. RZ

Reply:
Added.